sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-modules/files/arm64_defconfig-6.1

```diff
@@ -96,3 +96,10 @@ CONFIG_USB_ULPI=y
 CONFIG_USB_XHCI_PLATFORM=y
 CONFIG_VIRTUALIZATION=y
 CONFIG_XGENE_DMA=y
+# EXTRA CONFIG FOR DPAA2
+CONFIG_FSL_DPAA2_ETH=y
+CONFIG_FSL_DPAA2_ETH_DCB=y
+CONFIG_FSL_DPAA2_PTP_CLOCK=y
+CONFIG_PTP_1588_CLOCK_QORIQ=y
+CONFIG_FSL_MC_BUS=y
+CONFIG_FSL_MC_UAPI_SUPPORT=y
```

Comment:
This build failed with a following message:

```
 * ERROR: sys-kernel/coreos-modules-6.1.58::coreos failed (configure phase):
 *   Requested options not enabled in build:
 *     CONFIG_FSL_DPAA2_ETH
 *     CONFIG_FSL_DPAA2_ETH_DCB
 *     CONFIG_FSL_DPAA2_PTP_CLOCK
 *     CONFIG_FSL_MC_BUS
 *     CONFIG_FSL_MC_UAPI_SUPPORT
 *     CONFIG_PTP_1588_CLOCK_QORIQ
```

This usually means that some of these configs depend on something that was not enabled (either explicitly through our configs or implicitly by selected by some other enabled config or something). From what I have tracked these are the configs that dpaa2 requires directly or indirectly:

```
FSL_DPAA2_ETH
FSL_DPAA2_ETH_DCB
FSL_DPAA2_PTP_CLOCK
PTP_1588_CLOCK_QORIQ
FSL_MC_BUS
FSL_MC_UAPI_SUPPORT
FSL_MC_DPIO
DCB
PTP_1588_CLOCK
NET
POSIX_TIMERS
OF
ARCH_NXP
ARCH_LAYERSCAPE
```

From those, `DCB` and `NET` are already enabled in commonconfig. `POSIX_TIMERS`, `PTP_1588_CLOCK` are (implicitly?) enabled for both arm64 and amd64, and `OF` for `arm64`. You can see the final configs used for the kernel in our bincache: [amd64](http://bincache.flatcar-linux.net/images/amd64/3760.0.0+nightly-20231017-2100/flatcar_production_image_kernel_config.txt), [arm64](https://bincache.flatcar-linux.net/images/arm64/3760.0.0+nightly-20231017-2100/flatcar_production_image_kernel_config.txt)).

This would mean that `FSL_MC_DPIO`, `ARCH_NXP` and `ARCH_LAYERSCAPE` are missing from arm64 config. Also, please keep the file sorted (you can use the `sort_config.sh` script which is in the same directory as the configs you modified.

Reply:
Hello, thank you for the review. The initial configs were taken from a 5.10 Ubuntu, and the 6.1 does not have some of those anymore, trying now to see which ones are which, as for example, Flatcar latest stable has the 5.15 and the latest alpha has 6.1, and thus need to be different. What I am trying to fix is support for the DPAA2 network drivers for Layerscale LX2160A using PXE boot via the u-boot bootloader (somewhat exotic in nature). This commit is a working one https://github.com/ader1990/scripts/commit/23a078ad2f9488897c4204180e625eee6bf47659, rebased on latest stable -> flatcar-3602 branch.

I will need to update the above commit for the 6.1, work in progress.

Thank you.